app/main/services/process_request_json.py

```diff
@@ -16,9 +16,53 @@ def process_values_for_matching(values):
     return values
 
 
+def _ensure_value_list(json_string_or_list):
+    if isinstance(json_string_or_list, list):
+        return json_string_or_list
+    else:
+        return [json_string_or_list]
+
+
+def _append_conditionally(arguments, document):
+    """
+    A transformation processor that generates new field values in "target field" when
+    certain values are present in "field". The example use case is when
+    we are adding parent categories, whenever any one of their subcategories
+    is present.
+    :param arguments: dict -- the parameters to the processor as specified in configuration
+    :param document: dict -- the Elasticsearch document that we are transforming
+    """
+    source_field = arguments['field']
+    target_field = arguments.get('target_field') or source_field
+
+    if source_field in document:
+        source_values = _ensure_value_list(document[source_field])
+        source_values_set = set(source_values)
+        target_values = _ensure_value_list(document[target_field])
+
+        if any(value in source_values_set for value in arguments['any_of']):
```

Comment:
Could simply this to: `if source_values_set.intersection(arguments['any_of'])` which would be quicker to intuit the meaning of (to me!).